### PR TITLE
(SIMP-9659) Add scheduling options

### PR DESCRIPTION
**Pull request by trevor-vaughan**
- Changed
  - Default to `systemd` for scheduling
  - Use `--check` instead of `-C` by default to match what most scanners expect
  - Randomize the minute field so that I/O load is reduced on hosting platforms
- Added
  - Allow users to choose between the following options:
    - `systemd` => Default - Use systemd timers
    - `root`    => Legacy - Update root's crontab
    - `etc`     => Scanner Compat - Update /etc/crontab using Augeas
  - When one method is chosen, the others will be disabled

SIMP-9659 #close
SIMP-9261 #close

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
This will need to be pushed to remove the Puppet 5 tests  - Check the GitLab tests

File: manifests/init.pp [hunk 1]
```diff
@@ -99,32 +114,34 @@
...
+  Array[String]                            $report_urls       = [ 'file:@@{LOGDIR}/aide.report'],
+  Stdlib::Absolutepath                     $ruledir           = '/etc/aide.conf.d',
+  Variant[Hash,Array[String]]              $rules             = {},
+  Boolean                                  $enable            = false,
+  Optional[Simplib::Cron::Minute]          $minute            = undef,
+  Optional[Simplib::Cron::Hour]            $hour              = undef,
+  Optional[Simplib::Cron::Monthday]        $monthday          = undef,
+  Optional[Simplib::Cron::Month]           $month             = undef,
+  Optional[Simplib::Cron::Weekday]         $weekday           = undef,
+  Optional[Enum['root', 'etc', 'systemd']] $cron_method       = undef,
```
Comment (lnemsick-simp) [inline manifests/init.pp:134]: Which cron method is being selected?  Not clear here. Without a default, seems like no cron of any type will be run.

File: manifests/set_schedule.pp [hunk 2]
```diff
@@ -1,33 +1,156 @@
...
+  $_service = @("EOM")
+  [Service]
+  Type=oneshot
+  # Exit codes that simply mean that something was changed or updated but that
+  # the check ran successfully
+  SuccessExitStatus=1 2 3 4 5 6 7
+  ExecStart=$command
+  EOM
+
+  systemd::timer { 'puppet_aide.timer':
```
Comment (lnemsick-simp) [inline manifests/set_schedule.pp:71]: This resource is being created even when the user selects another option.  I understand that this systemd timer is disabled if not using 'systemd' cron method, but other design pattern of resource ensure absent is being used the rest of the file.  Why was this method used instead? Does it not apply to the system:timer resource?

File: CHANGELOG [hunk 3]
```diff
@@ -1,3 +1,17 @@
+* Thu Jun 10 2021 Trevor Vaughan <tvaughan@onyxpoint.com> - 6.5.0
+- Changed
+  - Default to `systemd` for scheduling
+  - Use `--check` instead of `-C` by default to match what most scanners expect
+  - Randomize the minute field so that I/O load is reduced on hosting platforms
+- Added
```
Comment (lnemsick-simp) [inline CHANGELOG:6]: Should indicate that it adds Puppet 7 support

**Review by lnemsick-simp — CHANGES_REQUESTED** [review]
IMHO the cron defaults should remain in the init.pp.
Did **not** deep-dive into the augeas rules as I have no domain knowledge. Am relying on the corresponding acceptance tests.

**Review by lnemsick-simp — APPROVED** [review]
Review comments largely addressed

**Pull request merged**